src/digraph.rs

```diff
@@ -2167,3 +2169,29 @@ fn weight_transform_callable(
         None => Ok(value.clone_ref(py)),
     }
 }
+
+#[pyproto]
```

Comment:
Maybe a short docstring, so it's obvious that this is for garbage collection?

Reply:
I added some comments for this in: https://github.com/Qiskit/retworkx/pull/258/commits/32f7c64d00ea6e59abc324666d3a3adee7d2009a (I didn't use docstrngs ,`///` because these don't get really exposed to the user so we don't need to include it in the exported objects)